cloudinit/sources/DataSourceCloudCIX.py

```diff
@@ -0,0 +1,206 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+import logging
+from typing import Optional
+
+from cloudinit import dmi, net, sources, url_helper, util
+from cloudinit.net.dhcp import NoDHCPLeaseError
+from cloudinit.net.ephemeral import EphemeralIPNetwork
+
+LOG = logging.getLogger(__name__)
+
+METADATA_URLS = ["http://169.254.169.254"]
+METADATA_VERSION = 1
+
+BUILTIN_DS_CONFIG = {
+    "retries": 3,
+    "timeout": 5,
+    "wait": 5,
+}
+
+
+CLOUDCIX_DMI_NAME = "CloudCIX"
+
+
+class DataSourceCloudCIX(sources.DataSource):
+
+    dsname = "CloudCIX"
+
+    def __init__(self, sys_cfg, distro, paths):
+        super(DataSourceCloudCIX, self).__init__(sys_cfg, distro, paths)
+        self.distro = distro
+        self.ds_cfg = util.mergemanydict(
+            [
+                util.get_cfg_by_path(sys_cfg, ["datasource", "CloudCIX"], {}),
+                BUILTIN_DS_CONFIG,
+            ]
+        )
+        self._metadata_url = None
+        self._net_cfg = None
+        self.url_retries = self.ds_cfg.get("retries")
+        self.url_timeout = self.ds_cfg.get("timeout")
+        self.url_sec_between_retries = self.ds_cfg.get("wait")
```

Comment:
Can drop a lot of this now
```suggestion
        self._metadata_url = None
        self._net_cfg = None
```

Reply:
Done.